app/main/helpers/frameworks/validation.py

```diff
@@ -0,0 +1,178 @@
+            for field in self.email_validation_fields:
+                if not re.match(EMAIL_REGEX, self.answers.get(field, '')):
+                    errors_map[field] = 'invalid_format'
+        return errors_map
+
+
+class EmptyValidator(FrameworkValidator):
+    """
+    An empty validator that will always pass.
+    """
+    pass
+
+
+class G7Validator(FrameworkValidator):
+    """
+    Validator for G-Cloud 7.
+    """
+    optional_fields = G7_OPTIONAL_FIELDS
+    email_validation_fields = G7_EMAIL_VALIDATION_FIELDS
+    character_limit = G7_CHARACTER_LIMIT
+
+    def errors(self):
+        errors_map = {}
+        errors_map.update(self.character_limit_errors())
+        errors_map.update(self.formatting_errors(self.answers))
+        errors_map.update(self.answer_required_errors())
+        return errors_map
+
+    def answer_required_errors(self):
+        required_fields = self.required_fields()
+        filled_fields = self.fields_with_values()
+        errors_map = {}
+
+        for field in required_fields - filled_fields:
+            errors_map[field] = 'answer_required'
+
+        return errors_map
+
+    def required_fields(self):
+        required_fields = set(self.all_fields())
+        #  Remove optional fields
+        if G7Validator.optional_fields is not None:
+            required_fields -= G7Validator.optional_fields
+
+        #  If you answered other to question 19 (trading status)
+        if self.answers.get('SQ1-1ci') == 'other (please specify)':
+            required_fields.add('SQ1-1cii')
+
+        #  If you answered yes to question 27 (non-UK business registered in EU)
+        if self.answers.get('SQ1-1i-i', False):
+            required_fields.add('SQ1-1i-ii')
+
+        #  If you answered 'licensed' or 'a member of a relevant organisation' in question 29
+        answer_29 = self.answers.get('SQ1-1j-i', [])
+        if len(answer_29) > 0 and \
+                ('licensed' in answer_29 or 'a member of a relevant organisation' in answer_29):
+                required_fields.add('SQ1-1j-ii')
+
+        # If you answered yes to either question 53 or 54 (tax returns)
+        if self.answers.get('SQ4-1a', False) or self.answers.get('SQ4-1b', False):
+            required_fields.add('SQ4-1c')
+
+        # If you answered Yes to questions 39 - 51 (discretionary exclusion)
+        dependent_fields = [
+            'SQ2-2a', 'SQ3-1a', 'SQ3-1b', 'SQ3-1c', 'SQ3-1d', 'SQ3-1e', 'SQ3-1f', 'SQ3-1g',
+            'SQ3-1h-i', 'SQ3-1h-ii', 'SQ3-1i-i', 'SQ3-1i-ii', 'SQ3-1j'
+        ]
+        if any(self.answers.get(field) for field in dependent_fields):
+            required_fields.add('SQ3-1k')
+
+        # If you answered No to question 26 (established in the UK)
+        if 'SQ5-2a' in self.answers and not self.answers['SQ5-2a']:
+            required_fields.add('SQ1-1i-i')
+            required_fields.add('SQ1-1j-i')
+
+        return required_fields
+
+validators['g-cloud-7'] = G7Validator
```

Comment:
`VALIDATORS = { 'g-cloud-7': G7Validator }` should make it clear that it's a constant and shouldn't be modified.